circuits/cpp/src/aztec3/circuits/abis/c_bind.cpp

```diff
@@ -459,7 +460,14 @@ WASM_EXPORT void abis__compute_transaction_hash(uint8_t const* tx_request_buf, u
     NT::fr::serialize_to_buffer(to_write, output);
 }
 
-WASM_EXPORT void abis__compute_call_stack_item_hash(uint8_t const* call_stack_item_buf, uint8_t* output)
+WASM_EXPORT void abis__compute_private_call_stack_item_hash(uint8_t const* call_stack_item_buf, uint8_t* output)
+{
+    CallStackItem<NT, PrivateTypes> call_stack_item;
+    serialize::read(call_stack_item_buf, call_stack_item);
+    NT::fr::serialize_to_buffer(call_stack_item.hash(), output);
```

Comment:
I notice the public counterpart to this (the function directly below this one) calls `get_call_stack_item_hash()`. Do we want to copy that approach here?

Reply:
The only reason why that's needed is because public call stack items actually represent two different things: a nested execution or an execution request. The `get_call_stack_item_hash` is there to process that two scenarios. But that doesn't apply to private.